### PR TITLE
Add clear DagRun endpoint.

### LINKED ISSUE #23419: Add a DAG Run clear endpoint to REST API
### Description

Add a POST endpoint to clear a dag run like we currently do [here](https://github.com/apache/airflow/blob/main/airflow/www/views.py#L2087).

Url format: `api/v1/dags/{dag_id}/dagRuns/{dag_run_id}/clear`

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tirkarthi**
closes: #23419
related: #23419

### CODE REVIEW

File: airflow/api_connexion/endpoints/dag_run_endpoint.py [hunk 1]
```diff
@@ -312,3 +317,54 @@ def update_dag_run_state(*, dag_id: str, dag_run_id: str, session: Session = NEW
...
+    except ValidationError as err:
+        raise BadRequest(detail=str(err))
+
+    dry_run = post_body.get('dry_run', False)
+    dag = current_app.dag_bag.get_dag(dag_id)
+    start_date = dag_run.logical_date
+    end_date = dag_run.logical_date
+
+    if dry_run:
+        task_instances = dag.clear(
```
Comment (Bowrna) [inline airflow/api_connexion/endpoints/dag_run_endpoint.py:348]: @tirkarthi I have some doubts about this part. How does this dag clear work without passing the dag_run_id? I am curious to know as I am working on a similar issue https://github.com/apache/airflow/issues/23227
Thank you

File: airflow/api_connexion/endpoints/dag_run_endpoint.py [hunk 2]
```diff
@@ -312,3 +317,54 @@ def update_dag_run_state(*, dag_id: str, dag_run_id: str, session: Session = NEW
...
+)
+@provide_session
+def clear_dag_run(*, dag_id: str, dag_run_id: str, session: Session = NEW_SESSION) -> APIResponse:
+    """Clear a dag run."""
+    dag_run: Optional[DagRun] = (
+        session.query(DagRun).filter(DagRun.dag_id == dag_id, DagRun.run_id == dag_run_id).one_or_none()
+    )
+    if dag_run is None:
+        error_message = f'Dag Run id {dag_run_id} not found in dag {dag_id}'
+        raise NotFound(error_message)
```
Comment (bbovenzi) [inline airflow/api_connexion/endpoints/dag_run_endpoint.py:336]:
```suggestion
        raise NotFound('DAG Run not found', detail=error_message)
```
I think we want the message to be the detail

File: airflow/api_connexion/endpoints/dag_run_endpoint.py [hunk 3]
```diff
@@ -312,3 +317,54 @@ def update_dag_run_state(*, dag_id: str, dag_run_id: str, session: Session = NEW
...
+            start_date=start_date,
+            end_date=end_date,
+            task_ids=None,
+            include_subdags=True,
+            include_parentdag=True,
+            only_failed=False,
+            dry_run=True,
+        )
+        return task_instance_reference_collection_schema.dump(
+            TaskInstanceReferenceCollection(task_instances=task_instances)
```
Comment (bbovenzi) [inline airflow/api_connexion/endpoints/dag_run_endpoint.py:358]: We should remove `execution_date` from the schema here, but that's out-of-scope for this PR.

**Review by bbovenzi — APPROVED** [review]
testing with postman. ~~other than one nit~~, lgtm

File: airflow/api_connexion/endpoints/dag_run_endpoint.py (hunk 2 above)
Comment (bbovenzi) [reply airflow/api_connexion/endpoints/dag_run_endpoint.py:336]: Nevermind, I see we do this all over the place.

**Pull request merged**